File: app/src/main/java/cjx/com/diary/view/fragment/FindFragment.java
package cjx.com.diary.view.fragment;

import android.graphics.Bitmap;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v4.graphics.drawable.RoundedBitmapDrawable;
import android.support.v4.graphics.drawable.RoundedBitmapDrawableFactory;
import android.support.v4.widget.SwipeRefreshLayout;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.Toolbar;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.request.target.BitmapImageViewTarget;
import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;

import java.util.ArrayList;
import java.util.List;

import bear.com.domain.model.News;
import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.Unbinder;
import cjx.com.diary.R;
import cjx.com.diary.base.BaseFragment;
import cjx.com.diary.presenter.impl.FindPresenterImp;
import cjx.com.diary.util.Utils;
import cjx.com.diary.util.miscreenshot.ScreenShotHelper;

/**
 * Created by bear on 2017/4/27.
 */

public class FindFragment extends BaseFragment {
    @BindView(R.id.iv_back)
    ImageView mBackIv;
    @BindView(R.id.tv_title)
    TextView mTitleTv;
    @BindView(R.id.tv_extend)
    TextView mExtendTv;

    public static Fragment newInstance() {
        return new FindFragment();
    }

    @BindView(R.id.sw_layout)
    SwipeRefreshLayout mSwl;
    @BindView(R.id.tb_title_bar)
    Toolbar tbTitleBar;
    @BindView(R.id.recycle_view)
    RecyclerView recycleView;
    Unbinder unbinder;
    private View scrollView;

    private FindPresenterImp mPresenter;

    private MyAdapter mAdapter;
    List<News> mData = new ArrayList<>();

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View contView = inflater.inflate(R.layout.frag_find, container, false);
        unbinder = ButterKnife.bind(this, contView);
        mPresenter = new FindPresenterImp();
        mPresenter.bindView(this, null);
        return contView;
    }

    public void setRefresh(boolean refresh) {
        if (mSwl != null) {
            mSwl.setRefreshing(refresh);
        }
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        mBackIv.setVisibility(View.GONE);
        mTitleTv.setText("趣闻");

        mExtendTv.setVisibility(View.VISIBLE);
        mExtendTv.setText("保存截图");
        mExtendTv.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                screenShot();
            }
        });


        mSwl.setOnRefreshListener(() -> {
            mPresenter.getQiuBai();
        });
        recycleView.setLayoutManager(new LinearLayoutManager(mActivity, LinearLayoutManager.VERTICAL, false));
        mAdapter = new MyAdapter(mData);
        recycleView.setAdapter(mAdapter);
        mPresenter.getQiuBai();
    }

    private void screenShot() {
        scrollView = ScreenShotHelper.getCanScrollView((ViewGroup) getView());
        ScreenShotHelper.screenShot(mActivity, scrollView, (bitmap, filePath) -> mActivity.runOnUiThread(() -> Utils.showToast(mActivity, "保存成功！")));
    }

    public void onRefresh(List<News> tem) {
        if (mActivity == null || mActivity.isFinishing()) return;
        mSwl.setRefreshing(false);
        if (null != tem && tem.size() > 0) {
            mData.clear();
            mData.addAll(tem);
            mAdapter.notifyDataSetChanged();
        }
    }


    private class MyAdapter extends BaseQuickAdapter<News, BaseViewHolder> {
        public MyAdapter(List<News> data) {
            super(R.layout.item_find_images, data);
        }

        @Override
        protected void convert(BaseViewHolder baseViewHolder, News s) {
            ImageView mImageView = baseViewHolder.getView(R.id.image);
            Glide.with(mActivity).load(s.image_url).asBitmap().centerCrop().into(new BitmapImageViewTarget(mImageView) {
                @Override
                protected void setResource(Bitmap resource) {
                    RoundedBitmapDrawable circularBitmapDrawable =
                            RoundedBitmapDrawableFactory.create(mActivity.getResources(), resource);
                    circularBitmapDrawable.setCircular(true);
                    mImageView.setImageDrawable(circularBitmapDrawable);
                }
            });
            baseViewHolder.setText(R.id.tv_user_name, s.user_name);
            baseViewHolder.setText(R.id.tv_content, s.content);
        }
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        unbinder.unbind();
    }
}

File: app/src/main/java/cjx/com/diary/view/fragment/FindFragment.kt
package cjx.com.diary.view.fragment

import android.graphics.Bitmap
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.graphics.drawable.RoundedBitmapDrawableFactory
import android.support.v7.widget.LinearLayoutManager
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import bear.com.domain.model.News
import cjx.com.diary.R
import cjx.com.diary.base.BaseFragment
import cjx.com.diary.presenter.impl.FindPresenterImp
import cjx.com.diary.util.Utils
import cjx.com.diary.util.miscreenshot.ScreenShotHelper
import com.bumptech.glide.Glide
import com.bumptech.glide.request.target.BitmapImageViewTarget
import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.BaseViewHolder
import kotlinx.android.synthetic.main.frag_find.*
import kotlinx.android.synthetic.main.view_title_bar.*
import java.util.*

/**
 * Created by bear on 2017/4/27.
 */

class FindFragment : BaseFragment() {

    private var scrollView: View? = null

    private var mPresenter: FindPresenterImp? = null

    private var mAdapter: MyAdapter? = null
    var mData: MutableList<News> = ArrayList()

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val contView = inflater.inflate(R.layout.frag_find, container, false)
        mPresenter = FindPresenterImp()
        mPresenter?.bindView(this, null)
        return contView
    }

    fun setRefresh(refresh: Boolean) {
        if (sw_layout != null) {
            sw_layout.isRefreshing = refresh
        }
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        iv_back.visibility = View.GONE
        tv_title.text = "趣闻"
        tv_extend.visibility = View.VISIBLE
        tv_extend.text = "保存截图"
        tv_extend.setOnClickListener { screenShot() }


        sw_layout.setOnRefreshListener { mPresenter?.getQiuBai() }
        mAdapter = MyAdapter(mData)
        recycle_view.layoutManager = LinearLayoutManager(activity, LinearLayoutManager.VERTICAL, false)
        recycle_view.adapter = mAdapter
        mPresenter?.getQiuBai()
    }

    private fun screenShot() {
        scrollView = ScreenShotHelper.getCanScrollView(view as ViewGroup?)
        ScreenShotHelper.screenShot(activity, scrollView) { _, _ -> activity?.runOnUiThread { Utils.showToast(activity, "保存成功！") } }
    }

    fun onRefresh(tem: List<News>?) {
        if (activity == null || activity!!.isFinishing) return
        sw_layout.isRefreshing = false
        if (null != tem && tem.isNotEmpty()) {
            mData.clear()
            mData.addAll(tem)
            mAdapter?.notifyDataSetChanged()
        }
    }


    private inner class MyAdapter(data: List<News>) : BaseQuickAdapter<News, BaseViewHolder>(R.layout.item_find_images, data) {

        override fun convert(baseViewHolder: BaseViewHolder, s: News) {
            val mImageView = baseViewHolder.getView<ImageView>(R.id.image)
            Glide.with(activity).load(s.image_url).asBitmap().centerCrop().into<BitmapImageViewTarget>(object : BitmapImageViewTarget(mImageView) {
                override fun setResource(resource: Bitmap) {
                    val circularBitmapDrawable = RoundedBitmapDrawableFactory.create(activity!!.resources, resource)
                    circularBitmapDrawable.isCircular = true
                    mImageView.setImageDrawable(circularBitmapDrawable)
                }
            })
            baseViewHolder.setText(R.id.tv_user_name, s.user_name)
            baseViewHolder.setText(R.id.tv_content, s.content)
        }
    }


    companion object {

        fun newInstance(): Fragment {
            return FindFragment()
        }
    }
}
